nvflare/private/fed/client/communicator.py

```diff
@@ -223,42 +159,53 @@ def getTask(self, servers, project_name, token, ssid, fl_ctx: FLContext):
             A CurrentTask message from server
 
         """
-        task, retry = None, self.retry
-        with self.set_up_channel(servers[project_name]) as channel:
-            stub = fed_service.FederatedTrainingStub(channel)
-            while retry > 0:
-                try:
-                    start_time = time.time()
-                    task = stub.GetTask(_get_client_state(project_name, token, ssid, fl_ctx))
-                    # Clear the stopping flag
-                    # if the connection to server recovered.
-                    self.should_stop = False
-
-                    end_time = time.time()
-
-                    if task.task_name == SpecialTaskName.TRY_AGAIN:
-                        self.logger.debug(
-                            f"Received from {project_name} server "
-                            f" ({task.ByteSize()} Bytes). getTask time: {end_time - start_time} seconds"
-                        )
-                    else:
-                        self.logger.info(
-                            f"Received from {project_name} server "
-                            f" ({task.ByteSize()} Bytes). getTask time: {end_time - start_time} seconds"
-                        )
-                    return task
-                except grpc.RpcError as grpc_error:
-                    self.grpc_error_handler(servers[project_name], grpc_error, "getTask", verbose=self.verbose)
-                    excep = FLCommunicationError(grpc_error, "grpc_error:getTask")
-                    retry -= 1
-                    time.sleep(5)
-            if self.should_stop:
-                raise excep
+        start_time = time.time()
+        shareable = Shareable()
+        shared_fl_ctx = FLContext()
+        shared_fl_ctx.set_public_props(get_serializable_data(fl_ctx).get_all_public_props())
+        shareable.set_header(ServerCommandKey.PEER_FL_CONTEXT, shared_fl_ctx)
+        task_message = new_cell_message(
+            {
+                CellMessageHeaderKeys.TOKEN: token,
+                CellMessageHeaderKeys.SSID: ssid,
+                CellMessageHeaderKeys.PROJECT_NAME: project_name,
+            },
+            fobs.dumps(shareable),
```

Comment:
Double FOBS

Reply:
Same. There's no double fobs issue, just different layer to call fobs.